Use explicitly 'flex' as lexer - else openSuSE uses a call with 'flex -l' from a shell wrapper and compilation fails.
 
--- scripts/kconfig/Makefile
+++ scripts/kconfig/Makefile
@@ -154,6 +154,7 @@
 # object files used by all kconfig flavours
 common-objs	:= confdata.o expr.o lexer.lex.o menu.o parser.tab.o \
 		   preprocess.o symbol.o util.o
+LEX = flex
 
 $(obj)/lexer.lex.o: $(obj)/parser.tab.h
 HOSTCFLAGS_lexer.lex.o	:= -I $(srctree)/$(src)
